### ISSUE TITLE
Support Provider aliases

### ISSUE THREAD

**Issue by suzuki-shunsuke** [terraform-linters/tflint-ruleset-aws#331]
## Problem

Terraform supports defining multiple configurations for the same provider, and select which one to use on a per-resource or per-module basis.

https://www.terraform.io/language/providers/configuration#alias-multiple-provider-configurations

But tflint doesn't support this, so even if multiple configurations are defined only one configuration is used.
This raises a problem in Deep Checking.

## How to reproduce

```console
$ tflint -v
TFLint version 0.35.0
+ ruleset.aws (0.13.2)
```

.tflint.hcl

```hcl
plugin "aws" {
  enabled = true
  version = "0.13.2"
  source  = "github.com/terraform-linters/tflint-ruleset-aws"

  deep_check = true
}
```

main.tf

```tf
terraform {
  required_version = ">= 0.15"
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "4.0.0"
    }
  }
}

provider "aws" {
  region = "ap-northeast-1"
}

provider "aws" {
  region = "us-east-1"
  alias  = "us-east-1"
}

resource "aws_instance" "reverse_proxy_a01" {
  ami           = "ami-088da9557aae42f39"
  instance_type = "t3.micro"
  key_name      = "bootstrap"
}
```

```console
$ tflint
2 issue(s) found:

Error: "ami-088da9557aae42f39" is invalid AMI ID. (aws_instance_invalid_ami)

  on main.tf line 21:
  21:   ami           = "ami-088da9557aae42f39"

Error: "bootstrap" is invalid key name. (aws_instance_invalid_key_name)

  on main.tf line 23:
  23:   key_name      = "bootstrap"
```

This error occurs even if Key Pair and AMI exist in the region `ap-northeast-1`, because tflint gets resources from `us-east-1` using the alias `us-east-1`.

## Solution

* Change `Runner.AwsClient` to map `map[string]*Client` and creates clients per Provider Configuration
* Check resource's `provider` attribute and get the appropriate Client from map of clients

I tried to implement this, but it doesn't work well yet.

https://github.com/terraform-linters/tflint-ruleset-aws/pull/332

I'll appreciate if you help me.
It is difficult to debug because I don't know how to output the plugin log.

## Reference

* https://www.terraform.io/language/providers/configuration#alias-multiple-provider-configurations

**Comment by suzuki-shunsuke** [comment]
What's the problem?

https://github.com/terraform-linters/tflint-ruleset-aws/runs/5814975813?check_suite_focus=true

```
panic: open aws-sdk-go/models/apis/account/2021-02-01/api-2.json: no such file or directory

goroutine 1 [running]:
main.main()
	/home/runner/work/tflint-ruleset-aws/tflint-ruleset-aws/rules/models/generator/main.go:64 +0xb71
exit status 2
rules/models/models.go:3: running "go": exit status 1
Error: Process completed with exit code 1.
```

**Comment by suzuki-shunsuke** [comment]
Oh, I see. Thank you. I can see the log.

**Comment by suzuki-shunsuke** [comment]
Solved by https://github.com/terraform-linters/tflint-ruleset-aws/pull/342
